bench/workgen/workgen.cxx

```diff
@@ -248,6 +281,231 @@ WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn)
+        THROW("Error opening a session.");
+    }
+
+    WT_DECL_RET;
+    ContextInternal *icontext = _workload->_context->_internal;
+    char uri[BUF_SIZE];
+    uint64_t total_creates = 0;
+    uint32_t db_size = get_dir_size_mb(_wt_home);
+
+    // Initially we start creating tables if the database size is less than the create target.
+    bool creating = db_size < _workload->options.dynamic_create_target;
+
+    while (!stopping) {
+        /*
+         * If we have been creating tables, continue until we reach the create target size. If we
+         * have not been creating tables, begin to do so if the database size falls below the create
+         * trigger.
+         */
+        db_size = get_dir_size_mb(_wt_home);
+        if (creating) {
+            creating = db_size < _workload->options.dynamic_create_target;
+            if (!creating) {
+                VERBOSE(*_workload,
+                  "Stopped creating new tables. db_size now "
+                    << db_size << " MB has reached the create target of "
+                    << _workload->options.dynamic_create_target << " MB.");
+            }
+        } else {
+            creating = db_size < _workload->options.dynamic_create_trigger;
+            if (creating) {
+                VERBOSE(*_workload,
+                  "Started creating new tables. db_size now "
+                    << db_size << " MB has reached the create trigger of "
+                    << _workload->options.dynamic_create_trigger << " MB.");
+            }
+        }
+        if (!creating) {
+            sleep(_workload->options.dynamic_create_period);
+            continue;
+        }
+
+        for (int creates = 0; creates < _workload->options.dynamic_create_count;
+             ++creates, ++total_creates) {
+            snprintf(uri, BUF_SIZE, "table:test_dynamic_%" PRIu64, total_creates);
+
+            // Make sure that this table is not a part of static table set, continue if it is.
+            if (icontext->_tint.count(uri) > 0 || icontext->_dyn_tint.count(uri) > 0)
+                continue;
+
+            // Create the table.
+            if ((ret = session->create(session, uri, "key_format=S,value_format=S")) != 0) {
+                if (ret == EBUSY)
+                    continue;
+                THROW("Table create failed in start_dynamic_table_mgmt.");
+            }
+
+            // The dynamic table data structures are protected by a mutex.
+            {
+                const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
+
+                // Add the table into the list of dynamic set.
+                tint_t tint = icontext->_dyn_tint_last;
+                icontext->_dyn_tint[uri] = tint;
+                icontext->_dyn_table_names[tint] = uri;
+                icontext->_dyn_table_runtime[tint] = TableRuntime();
+                ++icontext->_dyn_tint_last;
+                VERBOSE(*_workload, "Created table and added to the dynamic set: " << uri);
```

Comment:
This will generate a LOT of output if we are creating multiple tables every few seconds.

Reply:
Verbose is already very verbose. It generates thousands of lines of output per second for all the operations that the threads are doing. This is a relatively much higher granularity. I went through all the verbose entries I added and have removed one that I think was redundant.

Reply:
Ah well, in that case I withdraw my suggestion!